cli/src/cluster_query.rs

```diff
@@ -1401,32 +1413,45 @@ pub fn process_ping(
         }
 
         let to = config.signers[0].pubkey();
-        lamports += 1;
-
-        let build_message = |lamports| {
-            let mut ixs = vec![system_instruction::transfer(
-                &config.signers[0].pubkey(),
-                &to,
-                lamports,
-            )];
-            if let Some(compute_unit_price) = compute_unit_price {
-                ixs.push(ComputeBudgetInstruction::set_compute_unit_price(
-                    *compute_unit_price,
-                ));
-            }
-            Message::new(&ixs, Some(&config.signers[0].pubkey()))
-        };
-        let (message, _) = resolve_spend_tx_and_check_account_balance(
+
+        let mut total_batch_lamports = 0;
+        let batch_messages: Vec<Message> = (0..batch_size)
+            .map(|_i| {
+                let mut ixs = vec![system_instruction::transfer(
+                    &config.signers[0].pubkey(),
+                    &to,
+                    lamports,
+                )];
+                if let Some(compute_unit_price) = compute_unit_price {
+                    ixs.push(ComputeBudgetInstruction::set_compute_unit_price(
+                        *compute_unit_price,
+                    ));
+                }
+                total_batch_lamports += lamports;
+                lamports += 1;
+                Message::new(&ixs, Some(&config.signers[0].pubkey()))
+            })
+            .collect();
+
+        /*let (message, _) = resolve_spend_tx_and_check_account_balance(
             rpc_client,
             false,
             SpendAmount::Some(lamports),
             &blockhash,
             &config.signers[0].pubkey(),
             build_message,
             config.commitment,
-        )?;
-        let mut tx = Transaction::new_unsigned(message);
-        tx.try_sign(&config.signers, blockhash)?;
+        )?;*/
```

Comment:
@pieceofr if you could take over this PR and properly resolve this, that would be great, thanks!